youtube_dl/extractor/generic.py

```diff
@@ -1817,6 +1817,11 @@ def filter_video(urls):
                     'url': new_url,
                 }
         if not found:
+            # Last resort: just look for anything that looks like a media URL
+            found = set(re.findall(
+                r'\b((?:https?:)?//[A-Za-z0-9.:[\]]+/[A-Za-z0-9\-\._~:/()*+,;=%]+\.(?:mp3|mp4|flv|m3u8|smil)(?:\?[A-Za-z0-9\-\._~:/[\]()*+,;=%]*)?)\b',
```

Comment:
Perhaps. My other idea was to scan the contents of `<script>` tags, `data-*` attributes and flashvars (passed in whichever manner) for URLs of this form. That still isn't ideal; were it applied to ComCarrCoff, it would match lots of spurious URLs (given that the embedded JSON contains the data for all the videos, not just the current one). On the other hand, if the generic extractor reaches this point, there's little else you can do anyway.